### PR TITLE
Replace if/else with case/when

### PR DESCRIPTION
**Pull request by sebastiandeutsch**
I've replaced two if/else with case/when in `find_type`and `ruby_array_to_c`.

### CODE REVIEW

File: lib/tensorflow/tensor.rb [hunk 1]
```diff
@@ -121,27 +121,30 @@ def dimension_finder(array)
   #
   def find_type(data)
     start = data if self.rank == 0
-    start = data.flatten[0]  if self.rank != 0 
+    start = data.flatten[0]  if self.rank != 0
```
Comment (nethsix) [inline lib/tensorflow/tensor.rb]: @sebastiandeutsch What do you think of double-spacing between the statement and if?

**Comment by nethsix** [comment]
@sebastiandeutsch @Arafatk Reviewed. Looks good! Left a comment about whitespace.

**Comment by Arafatk** [comment]
@nethsix @sebastiandeutsch Thanks guys.

File: lib/tensorflow/tensor.rb (hunk 1 above)
Comment (sebastiandeutsch) [inline lib/tensorflow/tensor.rb]: @nethsix I wouldn't do double indents.

**Pull request merged**